### PR TITLE
Filter inaccurate locations

### LINKED ISSUE #433: Filter inaccurate GPS
GPS locations are not guaranteed to be accurate. We should filter out any locations with poor accuracy (>= 100m horizontal, <= 0m horizontal, >= 30m vertical, <= 0m vertical).

cc @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by frederoni**
Fixes #433 - Filter inaccurate locations

@ericrwolfe @bsudekum  👀

### CODE REVIEW

File: MapboxCoreNavigation/RouteController.swift [hunk 1]
```diff
@@ -390,9 +390,15 @@ extension RouteController: CLLocationManagerDelegate {
...
         self.rawLocation = location
         
-        delegate?.routeController?(self, didUpdateLocations: [location])
+        if location.isQualified {
+            delegate?.routeController?(self, didUpdateLocations: [location])
+        }
         
         sessionState.pastLocations.push(location)
+        
+        guard location.isQualified else {
```
Comment (bsudekum) [inline MapboxCoreNavigation/RouteController.swift]: Could this guard be moved above addition above?

File: MapboxCoreNavigation/CLLocation.swift [hunk 2]
```diff
@@ -3,6 +3,12 @@ import CoreLocation
 
 extension CLLocation {
     
+    var isQualified: Bool {
+        return
+            0...100 ~= horizontalAccuracy &&
```
Comment (bsudekum) [inline MapboxCoreNavigation/CLLocation.swift]: If the wifi is off and the user is in an urban environment, will this ever be below 100m? Should we start higher and filter down slowly?

File: MapboxCoreNavigation/RouteController.swift (hunk 1 above)
Comment (frederoni) [inline MapboxCoreNavigation/RouteController.swift]: We still want to collect the location, even though it's not qualified. However, we could move down L:393

File: MapboxCoreNavigation/CLLocation.swift (hunk 2 above)
Comment (frederoni) [inline MapboxCoreNavigation/CLLocation.swift]: The two modes that we use, `kCLLocationAccuracyBest` and `kCLLocationAccuracyBestForNavigation` are not using wifi, perhaps for a cold start but wifi doesn't support that level of accuracy. However, we should still run the numbers.

**Pull request merged**